TS-sprint-6

numerouno-life opened the pull request:
Пришлось изменить кучу файлов из-за проблемы с названием пакетов

numerouno-life committed:
TS-sprint-6
Пришлось изменить кучу файлов из-за проблемы с названием пакетов

numerouno-life committed:
TS-sprint-6
Пришлось изменить кучу файлов из-за проблемы с названием пакетов

numerouno-life committed:
TS-sprint-6
Пришлось изменить кучу файлов из-за проблемы с названием пакетов
доработки по codeStyle

irek-valiev commented on src/tasktracker/taskdata/Subtask.java:
```diff
@@ -1,4 +1,4 @@
-package TaskTracker.taskData;
+package tasktracker.taskdata;
```
поменяй еще и сам пакет на tasktracker, а то импорт не совсем соответствует фактическому названию пакета TaskTracker

irek-valiev commented on src/tasktracker/manager/InMemoryHistoryManager.java:
```diff
@@ -0,0 +1,89 @@
+            node.prev.next = node.next;
+            node.next.prev = node.prev;
+        }
+        node.prev = null;
```
эти два действия можно не делать, после завершения метода про node забудется и не важно, что там внутри

irek-valiev commented on src/tasktracker/manager/InMemoryTaskManager.java:
```diff
@@ -1,28 +1,24 @@
+import tasktracker.taskdata.Epic;
+import tasktracker.taskdata.Subtask;
+import tasktracker.taskdata.Task;
 
```
нужно удалить из истории через метод remove(int id) удалить задачи/подзадачи/эпики в методах deleteAllTasks(), deleteAllEpics(), deleteAllEpics(), deleteTaskById(int id), deleteSubtaskById(int id), deleteEpicById(int id)

А еще увидел на строчке 93 наш косяк аж с 4 спринта))) Там сейчас проверка if (epic == null), а должно быть if (epic != null)

numerouno-life committed:
TS-sprint-6
изменил название пакета tasktracker на tracker
убрал сброс на null node.prev and node.next
добавил вызов HistoryManager.remove в методы удаления задач.

irek-valiev commented on src/tracker/manager/InMemoryTaskManager.java:
```diff
@@ -50,6 +55,9 @@ public void deleteAllSubtasks() {
     public void deleteAllEpics() {
+        for (Integer id : epics.keySet()) {
+            historyManager.remove(id);
+        }
```
нужен еще один цикл, в котором из истории еще будут удалены подзадачи

irek-valiev commented on src/tracker/manager/InMemoryTaskManager.java:
```diff
@@ -149,6 +159,7 @@ public void deleteSubtaskById(int id) {
 
     @Override
     public void deleteEpicById(int id) {
+        historyManager.remove(id);
```
еще и подзадачи нужно удалять из истории

numerouno-life committed:
TS-sprint-6
изменил название пакета tasktracker на tracker
убрал сброс на null node.prev and node.next
добавил вызов HistoryManager.remove в методы удаления задач.

numerouno-life commented:
забыл описать изменения....
